runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/content/ByteStreamJVM.kt

```diff
@@ -21,16 +21,26 @@ fun ByteStream.Companion.fromFile(file: File): ByteStream = file.asByteStream()
 /**
  * Create a [ByteStream] from a file
  */
-fun File.asByteStream(): ByteStream = FileContent(this)
+fun File.asByteStream(start: Long = 0, endInclusive: Long = -1): ByteStream {
+    require(start >= 0) { "start index $start cannot be negative" }
+    require(endInclusive == -1L || endInclusive >= start) {
+        "end index $endInclusive must be greater than or equal to start index $start"
+    }
+
+    val len = length()
+    require(endInclusive < len) { "end index $endInclusive must be less than file size $len" }
+
+    return FileContent(this, start, endInclusive)
```

Comment:
**Suggestion**: We've got `length` at this point so we could replace a `-1` with a real value. This would eliminate the need for special `-1` handling down in `FileContent` and collapse the length calculation into simpler logic.